stl/inc/compare

```diff
@@ -340,9 +341,503 @@ struct compare_three_way {
+                using _Floating_type = decay_t<_Ty1>;
+                using _Traits        = _Floating_type_traits<_Floating_type>;
+                using _Uint_type     = typename _Traits::_Uint_type;
+
+                auto _Left_uint  = _STD bit_cast<_Uint_type>(_Left);
+                auto _Right_uint = _STD bit_cast<_Uint_type>(_Right);
+
+                // 1. Ultra-fast path: equal representations are equivalent.
+                if (_Left_uint == _Right_uint) {
+                    return weak_ordering::equivalent;
+                }
+
+                // 2. Fold negative zero into positive zero.
+                constexpr _Uint_type _Negative_zero = _Traits::_Shifted_sign_mask;
+
+                if (_Left_uint == _Negative_zero) {
+                    _Left_uint = 0;
+                } else if (_Right_uint == _Negative_zero) { // The representations are known to be non-equal here.
+                    _Right_uint = 0;
+                }
+
+                // 3. Examine the sign bits.
+                const bool _Left_negative  = (_Left_uint & _Traits::_Shifted_sign_mask) != 0;
+                const bool _Right_negative = (_Right_uint & _Traits::_Shifted_sign_mask) != 0;
+
+                // 4. Fast path: after folding negative zero, any negative value is less than any positive value.
+                if (_Left_negative != _Right_negative) {
+                    return _Left_negative ? weak_ordering::less : weak_ordering::greater;
+                }
+
+                // 5. Clear the (identical) sign bits. We've already stored them for use at the end.
+                _Left_uint &= ~_Traits::_Shifted_sign_mask;
+                _Right_uint &= ~_Traits::_Shifted_sign_mask;
+
+                // 6. Fold all NaN values together.
+                // (The fact that _Infinity_plus_one represents a signaling NaN is irrelevant here.)
+                constexpr _Uint_type _Infinity_plus_one = _Traits::_Shifted_exponent_mask + 1;
+
+                if (_Left_uint > _Infinity_plus_one) {
+                    _Left_uint = _Infinity_plus_one;
+                }
+
+                if (_Right_uint > _Infinity_plus_one) {
+                    _Right_uint = _Infinity_plus_one;
+                }
+
+                // 7. Perform the final comparison, reversed for negative values.
+                // (For example, 1.5 is less than 2.5, but -1.5 is greater than -2.5.)
+                if (_Left_negative) {
+                    return static_cast<weak_ordering>(_Right_uint <=> _Left_uint);
+                } else {
+                    return static_cast<weak_ordering>(_Left_uint <=> _Right_uint);
+                }
+            } else if constexpr (_Strat == _St::_Three) {
+                return static_cast<weak_ordering>(compare_three_way{}(_Left, _Right));
+            } else {
+                static_assert(_Strat == _St::_Strong);
+                return static_cast<weak_ordering>(_STD strong_order(_Left, _Right));

```

Comment:
Since only ADL cases get here (per my comment on the concept) we could add a `void strong_order();` block and strike `_STD` here to dispatch directly.

Reply:
Done with an `/* ADL, throughput optimization */` comment. Also done for the `noexcept` expression in `_Choose()`.